app/scripts/components/exploration/components/dataset-selector-modal/index.tsx

```diff
@@ -0,0 +1,213 @@
+    padding-bottom: ${glsp(1)};
+    box-shadow: 0 -1px 0 0 ${themeVal('color.base-200a')};
+  }
+
+  ${ModalBody} {
+    height: 100%;
+    min-height: 0;
+    overflow-y: auto;
+    display: flex;
+    flex-flow: column;
+    padding-top: ${glsp(2)};
+    gap: ${glsp(1)};
+  }
+
+  ${ModalFooter} {
+    display: flex;
+    gap: ${glsp(1)};
+    align-items: center;
+    position: sticky;
+    bottom: ${glsp(-2)};
+    z-index: 100;
+    background-color: ${themeVal('color.base-50')};
+    box-shadow: 0 -1px 0 0 ${themeVal('color.base-200a')};
+    > .selection-info {
+      margin-right: auto;
+    }
+  }
+`;
+
+interface DatasetSelectorModalProps {
+  revealed: boolean;
+  close: () => void;
+}
+
+const allDatasets = rawAllDatasets.map(currentDataset => {
+  return {
+    ...currentDataset,
+    layers: currentDataset.layers.map(l => ({
+      ...l,
+      parentDataset: {
+        id: currentDataset.id,
+        name: currentDataset.name
+        // infoDescription: currentDataset.infoDescription,
+      }
+    }))
+  };
+});
+
+const datasetLayers = allDatasets
+  .flatMap((dataset) => dataset.layers)
+  .filter((d) => !d.analysis?.exclude);
+
+  // Filter elements in arr1 that are also included in arr2
+function countOverlap(arr1, arr2) {
+  const commonElements = arr1.filter(element => arr2.includes(element));
+  return commonElements.length;
+}
+  
+
+export function DatasetSelectorModal(props: DatasetSelectorModalProps) {
+  const { revealed, close } = props;
+
+  const [timelineDatasets, setTimelineDatasets] = useAtom(timelineDatasetsAtom);
+
+  // Store a list of selected datasets and only confirm on save.
+  const [selectedIds, setSelectedIds] = useState<string[]>(
+    timelineDatasets.map((dataset) => dataset.data.id)
+  );
+
+  useEffect(() => {
+    setSelectedIds(timelineDatasets.map((dataset) => dataset.data.id));
+  }, [timelineDatasets]);
+
+  const onCheck = useCallback((id) => {
+    setSelectedIds((ids) =>
+      ids.includes(id) ? ids.filter((i) => i !== id) : [...ids, id]
+    );
+  }, []);
+
+  const onConfirm = useCallback(() => {
+    // Reconcile selectedIds with datasets.
+    setTimelineDatasets(
+      reconcileDatasets(selectedIds, datasetLayers, timelineDatasets)
+    );
+    close();
+  }, [close, selectedIds, timelineDatasets, setTimelineDatasets]);
+
+  const controlVars = useBrowserControls({
+    sortOptions
+  });
+  const { taxonomies, sortField, sortDir, onAction } = controlVars;
+  const search = controlVars.search ?? '';
+
+  // Clear filters when the modal is revealed.
+  const firstRevealRef = React.useRef(true);
+
+  useEffect(() => {
+    if (revealed) {
+      if (firstRevealRef.current) {
+        firstRevealRef.current = false;
+        return;
+      }
+      onAction(Actions.CLEAR);
+    }
+  }, [revealed]);
+
+  // Filters are applies to the veda datasets, but then we want to display the
+  // dataset layers since those are shown on the map.
+  const displayDatasets = useMemo(
```

Comment:
🔧 : I think it would be a good idea to type the return value here and then we can update the `ModalContentComponentProps` to receive this proper type for `displayDatasets`

maybe...
```
interface DisplayDatasetLayerInfo {
    countSelectedLayers: any;
    featured?: boolean | undefined;
    id: string;
    name: string;
    infoDescription?: string | undefined;
    taxonomy: Taxonomy[];
    description: string;
    usage?: DatasetUsage[] | undefined;
    media?: Media | undefined;
    layers: DatasetLayer[];
    related?: RelatedContentData[] | undefined;
    disableExplore?: boolean | undefined;
}
```

But if this could extend a current type, even better. This way we could probably use the same type for the layer cards if need be?

Reply:
Typing is a great idea. If my code needs to be clarified, this dataset is only to display the datasets on the modal. (This data does not get saved into Atom.)! Which removes the necessities of properties such as `infoDescription`.